### DISCUSSION TITLE
Porting DIV0.ASM from the original Descent code

### DISCUSSION THREAD

**Discussion by erysdren** [dxx-rebirth/dxx-rebirth#666] [General]
Howdy! I'm looking to make an SDL2 port of an 2.5D game engine from 1999 called Portal2D. It uses the DIV0.ASM code from Descent's 1998 source code release, and I'm curious how ya'll worked around those divide-by-zero errors when making these Descent sourceports.

I have the engine successfully compiling for DOS via Open Watcom v2, but the main roadblock to making it work on a modern system is the DIV0 code. Any advice or help would be appreciated!

Thanks,
Jaycie

**Comment by vLKp** [comment]
I am sorry, but I do not think you will get a satisfactory answer here.  I see no sign of a `DIV0.ASM` anywhere in the history of this project, which suggests that if it was ever used by Descent, it was removed a very long time ago.  I only began contributing to Rebirth in the early 2010s.  @btb or @zicodxx were involved with the project before I was, and might be able to tell you more.  Git history shows that there were other early contributors, but I don't know how you could reach them.

**Comment by btb** [comment]
It seems to me like the div0 code was disabled before the Descent source code was ever released. If I'm not mistaken it's just something that would have been used in early stages of debugging.

**Comment by vLKp** [comment]
Some of the Descent math code has explicit checks to avoid a division by zero.  Perhaps the original developers disabled the div0 handler and made it a point to change all the sites that could take a zero divisor to include such a check, thereby preventing the invalid operation and removing the need for a recovery handler.  This might have led to some intermittent crashes until they found all the offending sites, but once fixed, it would be much more efficient than letting the recovery handler run and handle the trap.
